samples/cpp/benchmark_app/main.cpp

```diff
@@ -329,6 +329,27 @@ void fuse_mean_scale(ov::preprocess::PrePostProcessor& preproc, const benchmark_
 }
 }  // namespace
 
+/**
+ * Sentry class to ensure the resource releasing from CompiledModel.
+ * Release static oneDNN cache used for inference with release_memory
+ * method. To be called once at the end of inference
+ */
+class model_sentry {
+public:
+    model_sentry(ov::CompiledModel& compiledModel) :
+        my_model(compiledModel) {}
+
+    ~model_sentry() {
+        if (my_model) {
+            my_model.release_memory();
+        }
+    }
+
+private:
+    ov::CompiledModel& my_model;
```

Comment:
```suggestion
    ov::CompiledModel my_model;
```
The copy will be shallow since it's a shared ptr inside. The benefit is that the ref will not be invalid